contracts/AirDrop.sol

```diff
@@ -0,0 +1,186 @@
+pragma solidity 0.6.4;
+
+import "./interface/IBEP20.sol";
+import "./interface/ITokenHub.sol";
+import "./interface/IAirDrop.sol";
+import "./interface/IParamSubscriber.sol";
+import "./lib/SafeMath.sol";
+import "./lib/BytesToTypes.sol";
+import "./System.sol";
+import "./MerkleProof.sol";
+
+contract AirDrop is IAirDrop, IParamSubscriber, System {
+    using SafeMath for uint256;
+
+    string public constant sourceChainID = "Binance-Chain-Ganges";
+    address public approvalAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
+    bytes32 public override merkleRoot = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    bool public merkleRootAlreadyInit = false;
+
+    // This is a packed array of booleans.
+    mapping(bytes32 => bool) private claimedMap;
+
+    function isClaimed(bytes32 node) public view override returns (bool) {
+        return claimedMap[node];
+    }
+
+    function claim(
+        bytes32 tokenSymbol, uint256 amount,
+        bytes calldata ownerPubKey, bytes calldata ownerSignature, bytes calldata approvalSignature,
+        bytes32[] calldata merkleProof) external override {
```

Comment:
We should add `noReentrant` as the transfer may involve BNB

Reply:
do you mean the reentrancy protection? we have a `claimedMap` to prevent this

Reply:
But I still suggest we have it.

Reply:
fixed